cli/tools/compile.rs

```diff
@@ -84,29 +86,6 @@ pub async fn compile(
   let ts_config_for_emit = cli_options
     .resolve_ts_config_for_emit(deno_config::deno_json::TsConfigType::Emit)?;
   check_warn_tsconfig(&ts_config_for_emit);
-  let root_dir_url = resolve_root_dir_from_specifiers(
-    cli_options.workspace().root_dir(),
-    graph
-      .specifiers()
-      .map(|(s, _)| s)
-      .chain(
-        cli_options
-          .node_modules_dir_path()
-          .and_then(|p| ModuleSpecifier::from_directory_path(p).ok())
-          .iter(),
-      )
-      .chain(include_files.iter())
-      .chain(
-        // sometimes the import map path is outside the root dir
-        cli_options
-          .workspace()
-          .to_import_map_path()
-          .ok()
-          .and_then(|p| p.and_then(|p| url_from_file_path(&p).ok()))
-          .iter(),
-      ),
-  );
-  log::debug!("Binary root dir: {}", root_dir_url);
```

Comment:
We now figure out the root after populating the VFS. This makes it much more reliable.